src/hnsw.c

```diff
@@ -189,5 +190,44 @@ Datum l2sq_dist(PG_FUNCTION_ARGS)
 {
     ArrayType *a = PG_GETARG_ARRAYTYPE_P(0);
     ArrayType *b = PG_GETARG_ARRAYTYPE_P(1);
-    PG_RETURN_FLOAT4(calc_distance(a, b));
+    PG_RETURN_FLOAT4(array_dist(a, b, usearch_metric_l2sq_k));
+}
+
+PGDLLEXPORT PG_FUNCTION_INFO_V1(cos_dist);
+Datum       cos_dist(PG_FUNCTION_ARGS)
+{
+    ArrayType *a = PG_GETARG_ARRAYTYPE_P(0);
+    ArrayType *b = PG_GETARG_ARRAYTYPE_P(1);
+    PG_RETURN_FLOAT4(array_dist(a, b, usearch_metric_cos_k));
+}
+
+PGDLLEXPORT PG_FUNCTION_INFO_V1(ham_dist);
+Datum       ham_dist(PG_FUNCTION_ARGS)
+{
+    ArrayType *a = PG_GETARG_ARRAYTYPE_P(0);
+    ArrayType *b = PG_GETARG_ARRAYTYPE_P(1);
+    PG_RETURN_FLOAT4(array_dist(a, b, usearch_metric_hamming_k));
+}
+
+PGDLLEXPORT PG_FUNCTION_INFO_V1(ldb_generic_dist);
+Datum       ldb_generic_dist(PG_FUNCTION_ARGS)
+{
+    ArrayType    *a = PG_GETARG_ARRAYTYPE_P(0);
+    ArrayType    *b = PG_GETARG_ARRAYTYPE_P(1);
+    MemoryContext m = CurrentMemoryContext;
+    bool          return_null = false;
+    while(m) {
+        if(strncmp(m->name, "PortalContext", 13) == 0) {
+            return_null = true;
+        }
+        m = m->parent;
+    }
+
+    if(return_null) {
+        // todo:: figure out why the Executor Portal is calling this function after the index scan
```

Comment:
It would be better to fix this case before we merge this in. Can you factor this out into a separate PR?
for now, `<->` will be the l2sq distance and for all the others people have to explicitly write out `_dist` function name.